sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -296,7 +297,8 @@ def add_runner_options(parser):
 
     prepare_response = job_service.Prepare(
         beam_job_api_pb2.PrepareJobRequest(
-            job_name='job', pipeline=proto_pipeline,
+            job_name=options.view_as(GoogleCloudOptions).job_name or 'job',
```

Comment:
GoogleCloudOptions doesn't seem appropriate for this. Should we relocate the option?

Reply:
That was initially my preferred approach, but I don't see a way to do it without breaking all the existing Dataflow pipelines that set a job name. See #9724

Reply:
Would it be possible to replicate the option and deprecate it in `GoogleCloudOptions`?

FWIW this option also exists in the Flink runner and the job_name is already reflected in the pipeline (see https://github.com/apache/beam/pull/9752#issuecomment-541250003), so what is the effect of this change going to be?

Reply:
`PipelineOptions` does not allow the same option to belong to multiple subclasses.

This PR changes `job_name` in the job request, not the pipeline options. e.g. here https://github.com/apache/beam/blob/master/runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/InMemoryJobService.java#L124

Reply:
I see, what is the user facing impact?

Regarding `PipelineOptions`, can it not be changed to allow for the co-existence of `job_name` in both classes for a transition period? @robertwb

Reply:
Yes, we could add an explicit `@property` in `GoogleCloudOptions` that delegates to the one in `PipelineOptions`.

Reply:
It's a good idea. Unfortunately, it looks like the override of `__settatr__` would take precedence over the property's setter. I'm not too familiar with how all these Python internals work, though, so I could be missing something obvious?

https://github.com/apache/beam/blob/16b4afa4469ee2d2fd5a5f4aafd77a156ee14e7b/sdks/python/apache_beam/options/pipeline_options.py#L336

Reply:
Huh. Perhaps override `_visible_option_list`?